src/main/java/commands/PlanCommand.java

```diff
@@ -1,11 +1,145 @@
 package commands;
 
+import data.plans.InvalidPlanException;
+import data.plans.Plan;
+import data.plans.PlanList;
+import storage.FileManager;
+import werkit.UI;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 public class PlanCommand extends Command {
-    public PlanCommand(String userInput) {
+    public static final String BASE_KEYWORD = "plan";
+    public static final String CREATE_ACTION_KEYWORD = "/new";
+    public static final String CREATE_ACTION_WORKOUTS_KEYWORD = "/workouts";
+    public static final String LIST_ACTION_KEYWORD = "/list";
+    public static final String DETAILS_ACTION_KEYWORD = "/details";
+    public static final String DELETE_ACTION_KEYWORD = "/delete";
+
+    private FileManager fileManager;
+    private UI ui = new UI();
+    private PlanList planList;
+
+    private String userAction;
+    private String userArguments;
+
+    private static Logger logger = Logger.getLogger(WorkoutCommand.class.getName());
+
+    public PlanCommand(String userInput, FileManager fileManager, PlanList planList,
+                       String userAction, String userArguments) throws InvalidCommandException {
         super(userInput);
+        this.fileManager = fileManager;
+        this.planList = planList;
+        setUserAction(userAction);
+        this.userArguments = userArguments;
     }
 
+    /**
+     * Gets the instance of the UI class.
+     *
+     * @return An instance of the UI class.
+     */
+    public UI getUI() {
+        return this.ui;
+    }
+
+    /**
+     * Gets the instance of the FileManager class.
+     *
+     * @return An instance of the FileManager class.
+     */
+    public FileManager getFileManager() {
+        return this.fileManager;
+    }
+
+    /**
+     * Gets the instance of the PlanList class.
+     *
+     * @return An instance of the PlanList class.
+     */
+    public PlanList getPlanList() {
+        return this.planList;
+    }
+
+    /**
+     * Gets the action of the workout command specified by the user.
+     *
+     * @return A string containing the action specified by the user.
+     */
+    public String getUserAction() {
+        return this.userAction;
+    }
+
+    /**
+     * Checks for the validity of the action specified by the user and sets the class field
+     * userAction to the action (if valid).
+     *
+     * @param userAction A string containing the action specified by the user.
+     * @throws InvalidCommandException If the action specified by the user is invalid.
+     */
+    public void setUserAction(String userAction) throws InvalidCommandException {
+        switch (userAction) {
+        case CREATE_ACTION_KEYWORD:
+        case LIST_ACTION_KEYWORD:
+        case DELETE_ACTION_KEYWORD:
+        case DETAILS_ACTION_KEYWORD:
+            this.userAction = userAction;
+            break;
+        default:
```

Comment:
Probably a //Fallthrough comment should be added 
From the [Java coding standard](https://se-education.org/guides/conventions/java/basic.html#layout) website:
The explicit `//Fallthrough` comment should be included whenever there is a case statement without a break statement.

Reply:
thank you!